signer/src/testing/dummy.rs

```diff
@@ -225,3 +239,79 @@ fn coinbase_tx<R: rand::RngCore + ?Sized>(
 
     coinbase_tx
 }
+
+/// Used to for fine-grained control of generating fake testing addresses.
+#[derive(Debug)]
+pub struct BitcoinAddresses(pub Range<usize>);
+
+impl fake::Dummy<BitcoinAddresses> for Vec<String> {
+    fn dummy_with_rng<R: rand::Rng + ?Sized>(config: &BitcoinAddresses, rng: &mut R) -> Self {
+        let num_addresses = config.0.clone().choose(rng).unwrap_or(1);
+        std::iter::repeat_with(|| secp256k1::Keypair::new_global(rng))
+            .take(num_addresses)
+            .map(|kp| {
+                let pk = bitcoin::CompressedPublicKey(kp.public_key());
+                Address::p2wpkh(&pk, Network::Regtest).to_string()
+            })
+            .collect()
+    }
+}
+
+impl<T> fake::Dummy<T> for WithdrawalAcceptEvent {
```

Comment:
I don't think we'd want to do a blanket implementation here. This prevents any other implementations for other fakers. Instead, I'd recommend specifying this to just the `fake::Faker` type:

```suggestion
impl fake::Dummy<fake::Faker> for WithdrawalAcceptEvent {
```

Reply:
Oh right 🤦🏽 good point.